File: ios/Classes/Module/WorkMomentsManager.swift
import Foundation
import OpenIMCore

public class WorkMomentsManager: BaseServiceManager {

  public override func registerHandlers() {
        super.registerHandlers()
        self["setWorkMomentsListener"] = setWorkMomentsListener
        self["getWorkMomentsUnReadCount"] = getWorkMomentsUnReadCount
        self["getWorkMomentsNotification"] = getWorkMomentsNotification
        self["clearWorkMomentsNotification"] = clearWorkMomentsNotification
    }

    func setWorkMomentsListener(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSetWorkMomentsListener(WorkMomentsListener(channel: channel))
        callBack(result)
    }

    func getWorkMomentsUnReadCount(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetWorkMomentsUnReadCount(BaseCallback(result: result), methodCall[string: "operationID"])
    }
    
    func getWorkMomentsNotification(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetWorkMomentsNotification(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[int: "offset"], methodCall[int: "count"])
    }
    
    func clearWorkMomentsNotification(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkClearWorkMomentsNotification(BaseCallback(result: result), methodCall[string: "operationID"])
    }
}

public class WorkMomentsListener: NSObject, Open_im_sdk_callbackOnWorkMomentsListenerProtocol {
    
    private let channel:FlutterMethodChannel

    init(channel:FlutterMethodChannel) {
        self.channel = channel
    }
    
    public func onRecvNewNotification() {
        CommonUtil.emitEvent(channel: self.channel, method: "workMomentsListener", type: "OnRecvNewNotification", errCode: nil, errMsg: nil, data: nil)
    }
    
}

File: android/src/main/java/io/openim/flutter_openim_sdk/manager/WorkMomentsManager.java
package io.openim.flutter_openim_sdk.manager;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.openim.flutter_openim_sdk.listener.OnBaseListener;
import io.openim.flutter_openim_sdk.listener.OnWorkMomentsListener;
import open_im_sdk.Open_im_sdk;

public class WorkMomentsManager extends BaseManager {

    public void setWorkMomentsListener(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.setWorkMomentsListener(new OnWorkMomentsListener());
    }

    public void getWorkMomentsUnReadCount(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getWorkMomentsUnReadCount(new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"));
    }

    public void getWorkMomentsNotification(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getWorkMomentsNotification(new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                int2long(methodCall, "offset"),
                int2long(methodCall, "count"));
    }

    public void clearWorkMomentsNotification(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.clearWorkMomentsNotification(new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"));
    }
}
